File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/PickMediaActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import pyxis.uzuki.live.richutilskt.utils.RPickMedia;
import pyxis.uzuki.live.richutilskt.utils.RichUtils;

public class PickMediaActivity extends BaseActivity {
    @BindView(R.id.gallery) Button gallery;
    @BindView(R.id.camera) Button camera;
    @BindView(R.id.video) Button video;
    @BindView(R.id.videoc) Button videoc;
    @BindView(R.id.txtUrl) TextView txtUrl;
    @BindView(R.id.imgView) ImageView imgView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pick);

        gallery.setOnClickListener(v -> RPickMedia.instance.pickFromGallery(this, this::resultMessage));

        camera.setOnClickListener(v -> RPickMedia.instance.pickFromCamera(this, this::resultMessage));

        video.setOnClickListener(v -> RPickMedia.instance.pickFromVideo(this, this::resultMessage));

        videoc.setOnClickListener(v -> RPickMedia.instance.pickFromVideoCamera(this, this::resultMessage));
    }

    private void resultMessage(int resultCode, String path) {
        if (resultCode == RPickMedia.PICK_FAILED) {
            throw new SecurityException("not granted permission to pick media files");
        }

        String realPath = RichUtils.getRealPath(Uri.parse(path), this);
        int width = RichUtils.getPhotoWidth(realPath);
        int height = RichUtils.getPhotoHeight(realPath);
        int degree = RichUtils.getPhotoOrientationDegree(realPath);

        txtUrl.setText("url -> " + realPath + ", width -> " + width + ", height ->" + height + ", degree -> " + degree);

        Bitmap bitmap = RichUtils.getBitmap(realPath);
        imgView.setImageBitmap(bitmap);

        RichUtils.requestMediaScanner(this, realPath);
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/PickMediaActivity.kt
package pyxis.uzuki.live.richutilssample

import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_pick.*
import pyxis.uzuki.live.richutilskt.utils.*

class PickMediaActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pick)
        
        gallery.setOnClickListener {
            RPickMedia.instance.pickFromGallery(this, { resultCode: Int, path: String -> resultMesage(resultCode, path) })
        }

        camera.setOnClickListener {
            RPickMedia.instance.pickFromCamera(this, { resultCode, path -> resultMesage(resultCode, path) })
        }

        video.setOnClickListener {
            RPickMedia.instance.pickFromVideo(this, { resultCode: Int, path: String -> resultMesage(resultCode, path) })
        }

        videoc.setOnClickListener {
            RPickMedia.instance.pickFromVideoCamera(this, { resultCode: Int, path: String -> resultMesage(resultCode, path) })
        }
    }

    private fun resultMesage(resultCode: Int, path: String) {
        if (resultCode == RPickMedia.PICK_FAILED) {
            throw SecurityException("not granted permission to pick media files")
        }

        val realPath = Uri.parse(path) getRealPath this@PickMediaActivity
        val width = realPath.getImageWidth()
        val height = realPath.getImageHeight()
        val degree = getPhotoOrientationDegree(realPath)

        txtUrl.text = "uri -> $realPath , width -> $width , height -> $height , degree -> $degree"

        val bitmap = BitmapFactory.decodeFile(realPath)
        imgView.setImageBitmap(bitmap)

        requestMediaScanner(realPath)
    }
}
